src/ddmd/hdrgen.d

```diff
@@ -574,13 +574,11 @@ public:
         buf.writenl();
         buf.writestring("finally");
         buf.writenl();
-        buf.writeByte('{');
-        buf.writenl();
-        buf.level++;
+        if (!s.finalbody.isScopeStatement())
+            buf.level++;
         s.finalbody.accept(this);
-        buf.level--;
-        buf.writeByte('}');
-        buf.writenl();
+        if (!s.finalbody.isScopeStatement())
+            buf.level--;
```

Comment:
I fully agree, but I was following existing conventions. Should I also update the other locations in the file that use this pattern?

I'll add a test case.